File: src/ruby/Ecosistema/Ecosistema.rb
require('./Ecosistema/Locacion.rb');
require('./Ecosistema/Biotopo.rb');
require('./Ecosistema/Biosenosis.rb');

module Ecosistema
		# @author galindo
public
	class Ecosistema

		# @ParamType aArgs String*
		# the command line arguments
		# @ReturnType void
		public
		def main(*aArgs)
			# Not yet implemented
		end

		# @ReturnType String
		public
		def getNombre()
			return @_nombre
		end

		# @ParamType aNombre String
		# @ReturnType void
		public
		def setNombre(aNombre)
			@_nombre = aNombre
		end

		public
		def initialize()
			# @AttributeType String
			@_nombre
			# @AttributeType ecosistema.Locacion
			# # @AssociationType ecosistema.Locacion
			# # @AssociationKind Aggregation
			@_locaciones
			# @AttributeType ecosistema.Biotopo
			# # @AssociationType ecosistema.Biotopo
			# # @AssociationMultiplicity *
			# # @AssociationKind Aggregation
			@_oranismosVivos = Array.new()
			# @AttributeType ecosistema.Biosenosis
			# # @AssociationType ecosistema.Biosenosis
			# # @AssociationMultiplicity *
			# # @AssociationKind Aggregation
			@_mediosFisicos = Array.new()

			# Not yet implemented
		end
	end
end

File: src/ecosistema/Ecosistema.java
package ecosistema;

import java.util.Vector;
import ecosistema.Biotopo;
import ecosistema.Biosenosis;

/**
 * @author galindo
 */
public class Ecosistema {

    private String _nombre;
    public Locacion _locaciones;
    public Vector<Biotopo> _oranismosVivos = new Vector<>();
    public Vector<Biosenosis> _mediosFIsicos = new Vector<>();

    public String getNombre() {
        return this._nombre;
    }

    public void setNombre(String aNombre) {
        this._nombre = aNombre;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Ecosistema{");
        sb.append("_nombre=").append(_nombre);
        sb.append(", _locaciones=").append(_locaciones);
        sb.append(", _oranismosVivos=").append(_oranismosVivos);
        sb.append(", _mediosFIsicos=").append(_mediosFIsicos);
        sb.append('}');
        return sb.toString();
    }
    
    
}
